erc20-migration/contracts/ZTESTZendBackupVault.sol

```diff
@@ -0,0 +1,127 @@
+        return balances[zendAddress];
+    }
+    
+    /// @notice Return the cumulativeHash  calculated so far
+    function getCumulativeHash() public view returns (bytes32) {
+        return _cumulativeHash;
+    }
+
+    /// @notice Set official ZEN ERC-20 smart contract that will be used for minting
+    function setERC20(address addr) public {
+        if (msg.sender != admin) revert UnauthorizedOperation();     
+        if (address(zenToken) != address(0)) revert UnauthorizedOperation();  //ERC-20 address already set
+        if(addr == address(0)) revert AddressNotValid();
+        zenToken = IERC20Mintable(addr);
+    }
+
+    /// @notice Claim a P2PKH balance.
+    ///         destAddress is the receiver of the founds
+    ///         hexSignature is the signature of the claiming message. Must be generated in a compressed format to claim a zend address
+    ///         generated with the public key in compressed format, or uncompressed otherwise.
+    ///         (Claiming message is predefined and composed by the string 'ZENCLAIM' concatenated with the destAddress in lowercase string hex format)
+    ///         pubKeyX and pubKeyY are the first 32 bytes and second 32 bytes of the signing key (we use always the uncompressed format here)
+    ///         Note: we pass the pubkey explicitly because the extraction from the signature would be GAS expensive.
+    function claimP2PKH(address destAddress, bytes memory hexSignature, bytes32 pubKeyX, bytes32 pubKeyY) public {
+        if (_cumulativeHash != cumulativeHashCheckpoint) revert CumulativeHashNotValid(); //Loaded data not matching - distribution locked 
+        if (address(zenToken) == address(0)) revert ERC20NotSet();
+        if (address(destAddress) == address(0)) revert AddressNotValid();
+
+        VerificationLibrary.Signature memory signature = VerificationLibrary.parseZendSignature(hexSignature);
+        bytes20 zenAddress;
+        if (signature.v == 31 || signature.v == 32){
+            //signature was compreesed, also the zen address will be from the compressed format
+             zenAddress = VerificationLibrary.pubKeyCompressedToZenAddress(pubKeyX, VerificationLibrary.signByte(pubKeyY));
+        }else{
+             zenAddress = VerificationLibrary.pubKeyUncompressedToZenAddress(pubKeyX, pubKeyY);
+        }
+        string memory asString = Strings.toHexString(uint256(uint160(destAddress)), 20);
+        string memory strMessageToSign = string(abi.encodePacked(MESSAGE_PREFIX, asString));
+        bytes32 messageHash = VerificationLibrary.createMessageHash(strMessageToSign);
+        VerificationLibrary.verifyZendSignature(messageHash, signature, pubKeyX, pubKeyY);
+
+        if (balances[zenAddress].distributed == true) revert AlreadyClaimed(zenAddress);
+        uint256 amount = balances[zenAddress].amount;
+        if(amount == 0) revert NothingToClaim(zenAddress);
+        zenToken.mint(destAddress, amount);
+        balances[zenAddress].distributed = true;
```

Comment:
```suggestion
        balances[zenAddress].distributed = true;
        zenToken.mint(destAddress, amount);
```
In general, the[ best practice](https://fravoll.github.io/solidity-patterns/checks_effects_interactions.html) mandates that calls to external contracts should happen at the very end of a method, and, in particular, no further state changes should happen after the external call. This guards against possible reentrancy attacks, reducing the attack surface of the contract. Of course, in this particular case, `zenToken` is a trusted contract, thus no reentrancy attack would actually be possible, but I think it's safer to stick to best practices.